src/server/frontend_wayland/virtual_pointer_v1.cpp

```diff
@@ -202,7 +202,12 @@ mf::VirtualPointerV1::VirtualPointerV1(
 
 mf::VirtualPointerV1::~VirtualPointerV1()
 {
-    ctx->device_registry->remove_device(pointer_device);
+    // Destruction must happen on the linearising_executor to prevent races with create_and_dispatch_event
```

Comment:
Ah! This is actually racing with `frame()` (which is what dispatches something to the `linearising_executor`)!

Reply:
Yup! That was the only race that I saw here from `VirtualPointer`'s perspective. I am also unsure if the rationale holds up, but I'll be curious to see if this prevents further breaking